cfme/tests/cloud/test_cloud_proxy.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+# These tests don't work at the moment, due to the security_groups multi select not working
+# in selenium (the group is selected then immediately reset)
+import pytest
+
+import cfme.fixtures.pytest_selenium as sel
+from cfme.cloud.provider import CloudProvider
+from utils import testgen
+from utils.log import logger
+from utils.path import proxy_path
+from utils.datafile import load_data_file as df
+from cfme.web_ui import flash, form_buttons
+from utils.appliance.implementations.ui import navigate_to
+
+
+pytestmark = [
+    pytest.mark.ignore_stream("upstream"),
+    pytest.mark.tier(1)
+]
+
+pytest_generate_tests = testgen.generate([CloudProvider], scope="module")
+
+
+def test_proxy_valid(appliance, provider, setup_provider, request):
+    logger.info("Begin Test Valid Proxy for Provider {}".format(provider.type))
+
+    if provider.type == "ec2":
+        data_file = df(str(proxy_path.join('ec2_proxy_settings_valid.data')))
+    elif provider.type == "openstack":
+        data_file = df(str(proxy_path.join('openstack_proxy.data')))
+    elif provider.type == "azure":
+        data_file = df(str(proxy_path.join('azure_proxy_settings_valid.data')))
+    elif provider.type == "gce":
+        data_file = df(str(proxy_path.join('gce_proxy.data')))
+
+    @request.addfinalizer
+    def _cleanup_proxy():
+        clean_up(appliance, df(str(proxy_path.join('clear_all_proxies.data'))))
+
+    appliance.ssh_client.run_command(
+        "echo \'{}\' > /var/www/miq/vmdb/config/settings.local.yml".format(data_file.read()))
+    restart_appliance(appliance)
+
+    login_page = navigate_to(appliance.server, 'LoginScreen')
+    login_page.login_admin()
+    navigate_to(provider, 'EditFromDetails')
+
+    sel.click(form_buttons.validate)
+    flash.assert_message_match('Credential validation was successful')
```

Comment:
There is a widget but it doesn't work.  Only navigates through some of the steps but dies when trying to find the quadicon.  Not really sure how to fix it, so I'm using the above until it is.

    def step(self):
>       sel.click(Quadicon(self.obj.name, self.obj.quad_name))
E       AttributeError: type object 'CloudProvider' has no attribute 'name'